oauth2-authorization-server/src/main/java/org/springframework/security/config/annotation/web/configuration/OAuth2AuthorizationServerConfiguration.java

```diff
@@ -47,12 +48,17 @@ public static void applyDefaultSecurity(HttpSecurity http) throws Exception {
 				new OAuth2AuthorizationServerConfigurer<>();
 		RequestMatcher endpointsMatcher = authorizationServerConfigurer
 				.getEndpointsMatcher();
+		RequestMatcher oidcClientRegistrationEndpointMatcher = authorizationServerConfigurer
+				.getOidcClientRegistrationEndpointMatcher();
 
 		http
 			.requestMatcher(endpointsMatcher)
 			.authorizeRequests(authorizeRequests ->
-				authorizeRequests.anyRequest().authenticated()
+						authorizeRequests.requestMatchers(oidcClientRegistrationEndpointMatcher)
```

Comment:
Revert back to `authorizeRequests.anyRequest().authenticated()` and perform the authorization check in `OidcClientRegistrationAuthenticationProvider`

Reply:
Fixed